File: 0121_best_time_to_buy_and_sell_stocks/one_pass/kotlin/solution.kt
fun maxProfit(prices: IntArray): Int {
    var minPrice = prices[0]
    var bestProfit = 0

    for (price in prices) {
        bestProfit = maxOf(bestProfit, price - minPrice)
        minPrice = minOf(minPrice, price)
    }

    return bestProfit
}

File: 0121_best_time_to_buy_and_sell_stocks/one_pass/go/solution.go
package best_time_to_buy_and_sell_stock

func maxProfit(prices []int) int {
	minPrice := prices[0]
	bestProfit := 0

	for _, price := range prices {
		if profit := price - minPrice; profit > bestProfit {
			bestProfit = profit
		}
		if price < minPrice {
			minPrice = price
		}
	}

	return bestProfit
}
